### ISSUE TITLE
Saving and loading trained ElephasTransformer

### ISSUE THREAD

**Issue by Gasolgasol** [maxpumperla/elephas#179]
I am trying to train ElephasEstimator in one file, save the resulting ElephasTransformer and then load it and do prediction/evaluation in another file. I use `fitted_model.save( "test.h5")` to save the transformer and then `transformer = load_ml_transformer('test.h5')` to load it. However while I try to run prediction "pred_test = transformer.transform(df)",  I get an error  `AttributeError: 'ElephasTransformer' object has no attribute 'weights'` . If I run prediction on a trained model right after training without saving and loading, I don't get any errors. Please help.

**Comment by danielenricocahall** [comment]
I will look into this over the weekend!

**Comment by OscarDPan** [comment]
oh yeah @Gasolgasol I was trying to do the same thing. In the nutshell the test.h5 doesn't include any information other than what the yaml contains.
```
        weights = training_rdd.ctx.broadcast(transformer.get_weights())
        transformer = ElephasTransformer(
            featuresCol=your_feat_col_name,
            labelCol=your_label_col_name,
            outputCol=your_output_col_name,
            weights=weights,
            custom_objects=custom_objects,
            keras_model_config=keras_transformer_str,
            loss=loss_func
        )```

**Comment by danielenricocahall** [comment]
Just released a patch (1.4.2) with the fix - we just had to ensure `weights` and `model_type` were available in the config for serialization/deserialization.

**Comment by Gasolgasol** [comment]
Thank you for fixing the bug. Instead of sitting and waiting I just decided to go ahead with RDDs and SparkModel approach for now. So testing ElephasTransformer is on hold :-)
